wagtail_localize/machine_translators/tests/test_libretranslate_translator.py

```diff
@@ -0,0 +1,93 @@
+from django.test import TestCase, override_settings
+from wagtail.models import Locale
+
+from wagtail_localize.machine_translators import get_machine_translator
+from wagtail_localize.machine_translators.libretranslate import LibreTranslator
+
+
+LIBRETRANSLATE_SETTINGS_ENDPOINT = {
+    "CLASS": "wagtail_localize.machine_translators.libretranslate.LibreTranslator",
+    "OPTIONS": {
+        "LIBRETRANSLATE_URL": "https://libretranslate.org",
+        "API_KEY": "test-api-key",
+    },
+}
+
+
+class TestLibreTranslator(TestCase):
+    @override_settings(
+        WAGTAILLOCALIZE_MACHINE_TRANSLATOR=LIBRETRANSLATE_SETTINGS_ENDPOINT
+    )
+    def setUp(self):
+        self.english_locale = Locale.objects.get()
+        self.french_locale = Locale.objects.create(language_code="fr")
+        self.translator = get_machine_translator()
+
+    def test_api_endpoint(self):
+        self.assertIsInstance(self.translator, LibreTranslator)
+        api_endpoint = self.translator.get_api_endpoint()
+        self.assertEqual(api_endpoint, "https://libretranslate.org")
+
+    # This probably requires a request to use the API but the test works against my local instance
```

Comment:
rather than making an actual API call, I'd test 2 things:

1. that `translate` will raise for status (so, mock the post call to that effect)
2. that we get a dict of `string: StringValue(translation)` from the response, if there are translated strings. Again, using a mocked response

Reply:
could I have a bit of pointer on mocking a post request please?